Manage expenses + workflow

gaschneider opened the pull request:
This is the PR that brings the entire system together, giving access to the expenses page, and running the workflow to track approvals and rejections.

Still missing the tests, will add it soon

gaschneider committed:
initial expense controller and create action with validation

gaschneider committed:
Merge remote-tracking branch 'origin/main' into gab/manage-expenses

gaschneider committed:
adiciona referencia a categoria em nivel de dispesas

gaschneider committed:
add create expense and expense page in frontend
adjust backend for expenses

gaschneider committed:
Create and list expenses(need to fix issue with creation and test list)

gaschneider committed:
Fix issue creating expense, add view modal for expenses. Fix styles

gaschneider committed:
Update workflow, update expense and expense status table with their respective migrations and generate new endpoints for the expense controller, update in view expense modal

gaschneider committed:
update expense hook

gaschneider committed:
workflow behavior, missing approval

gaschneider committed:
Fix filter and approval for expenses

gaschneider committed:
Add ability to update expenses on draft

gaschneider committed:
add comment to status, fix some UI issues

magicblak reviewed (changes requested):
1. Form validation: The expense form lacks front-end validation, allowing submission with empty values.
2. Comment persistence: Comments added during status transitions are not being saved in the database.
3. Table sorting: The table sorting functionality is not working correctly.

Aside from these issues, the rest of the workflow and expense registration is working as expected.

gaschneider committed:
Address PR comments

gaschneider commented:
> 1. Form validation: The expense form lacks front-end validation, allowing submission with empty values.
> 2. Comment persistence: Comments added during status transitions are not being saved in the database.
> 3. Table sorting: The table sorting functionality is not working correctly.
> 
> Aside from these issues, the rest of the workflow and expense registration is working as expected.

Addressed the issues above, also made Currency a single value enum for now, since we still need to address the currency in rules, issue #6 created for it

gaschneider committed:
fix department permission to view expenses

gaschneider committed:
Fix expense permissions frontend

magicblak reviewed (commented):
1. Users without permission to create expenses can still view the buttom to create.
2. Departments dropdown lists all departments, including those the user lacks permission to view (dropdown only, not data).
3. VIEW_EXPENSES permission grants access to the expenses page but doesn't display the list. APPROVE_EXPENSES is incorrectly required to view expenses

gaschneider committed:
Address PR comments

gaschneider commented:
> * Users without permission to create expenses can still view the buttom to create.
> * Departments dropdown lists all departments, including those the user lacks permission to view (dropdown only, not data).
> * VIEW_EXPENSES permission grants access to the expenses page but doesn't display the list. APPROVE_EXPENSES is incorrectly required to view expenses

Addressed the issues mentioned, should be working now.

magicblak reviewed (commented):
Department dropdown list in expenses page is not listing all departments to admin

gaschneider committed:
Fix departments for admin

gaschneider commented:
> Department dropdown list in expenses page is not listing all departments to admin

Addressed issue
